addons/portal/models/portal_mixin.py

```diff
@@ -132,3 +132,17 @@ def get_portal_url(self, suffix=None, report_type=None, download=None, query_str
             '#%s' % anchor if anchor else ''
         )
         return url
+
+    def _get_portal_data_from_context(self):
```

Comment:
This has nothing to do at model level, nobody is going to override this method (and overriding it would make no sense anyway), especially used once. Please keep data marshmalling where it is used, aka in mail.message as an inline function in your compute method.

Reply:
Actually this whole method should be removed, globally just do .get() + sudo(False) directly in compute.